include/onnxruntime/core/graph/graph.h

```diff
@@ -711,6 +712,12 @@ class Graph {  // NOLINT(clang-analyzer-optin.performance.Padding): preserve exi
    *    and replaces graph initializers with its content.
    */
   common::Status InjectExternalInitializedTensors(const InlinedHashMap<std::string, OrtValue>& external_initializers);
+
+  /** This function takes externally provided files in memory for initializers with external
+   *    data and replaces graph initializers with its content.
+   */
+  common::Status InjectExternalInitializersFromFilesInMemory(
+      const InlinedHashMap<ORT_STRING, std::pair<void*, size_t>>& external_initializer_files);
```

Comment:
can we stick to `std::basic_string<ORTCHAR_T>` instead of `ORT_STRING` given that it's just in one place? Besides, we should avoid using typedefs. The modern C++ way is to use the `using` keyword.

Reply:
OK. Yes, it makes it clear especially for the API header files

Reply:
I disagree. `PathString` is a well defined typedef we use internally. By saying typedef I meant `using` which is a more powerful modern typedef anyway.
For the API, I would also define such a string because we are using it in other APIs.

Reply:
It's hard to use it in some API header files like c_api.h, cxx_inline.h. PathString is defined in path_string.h which includes c_api.h.
I'll try to use PathString as much as possible.

Reply:
updated.